File: FinalBootMVC/src/main/java/com/springboot/pkg/model/Matiere.java
package com.springboot.pkg.model;

import java.util.Set;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.OneToMany;
import javax.persistence.SequenceGenerator;
import javax.persistence.Version;

@Entity
public class Matiere {

	@Id
	@SequenceGenerator(name = "matiereSeq", sequenceName = "matiere_seq", allocationSize = 1)
	@GeneratedValue(generator = "matiereSeq", strategy = GenerationType.SEQUENCE)
	private Long id;
	
	private String titre;
	private Integer duree;
	private String prerequis;
	private String contenu;
	private String objectif;

	@OneToMany(mappedBy = "matiere")
	private Set<Module> module;

	@OneToMany(mappedBy = "key.matiere")
	private Set<Competence> formateur;

	@Version
	private int version;

	public Matiere() {

	}

	public Matiere(String titre, Integer duree, String prerequis, String contenu, String objectif) {
		this.titre = titre;
		this.duree = duree;
		this.prerequis = prerequis;
		this.contenu = contenu;
		this.objectif = objectif;
	}

	public Long getId() {
		return id;
	}

	public void setId(Long id) {
		this.id = id;
	}

	public String getTitre() {
		return titre;
	}

	public void setTitre(String titre) {
		this.titre = titre;
	}

	public Integer getDuree() {
		return duree;
	}

	public void setDuree(Integer duree) {
		this.duree = duree;
	}

	public String getPrerequis() {
		return prerequis;
	}

	public void setPrerequis(String prerequis) {
		this.prerequis = prerequis;
	}

	public String getContenu() {
		return contenu;
	}

	public void setContenu(String contenu) {
		this.contenu = contenu;
	}

	public String getObjectif() {
		return objectif;
	}

	public void setObjectif(String objectif) {
		this.objectif = objectif;
	}

	public Set<Module> getModule() {
		return module;
	}

	public void setModule(Set<Module> module) {
		this.module = module;
	}

	public Set<Competence> getFormateur() {
		return formateur;
	}

	public void setFormateur(Set<Competence> formateur) {
		this.formateur = formateur;
	}

	public int getVersion() {
		return version;
	}

	public void setVersion(int version) {
		this.version = version;
	}

	@Override
	public int hashCode() {
		final int prime = 31;
		int result = 1;
		result = prime * result + ((id == null) ? 0 : id.hashCode());
		return result;
	}

	@Override
	public boolean equals(Object obj) {
		if (this == obj)
			return true;
		if (obj == null)
			return false;
		if (getClass() != obj.getClass())
			return false;
		Matiere other = (Matiere) obj;
		if (id == null) {
			if (other.id != null)
				return false;
		} else if (!id.equals(other.id))
			return false;
		return true;
	}

}

File: FinalAngular/src/app/model/matiere.ts
import {Competence} from './competence';
import {Module} from './module';
export class Matiere {

  private _id: number;
  private _module: Array<Module>;
  private _formateur: Array<Competence>;


  constructor(private _prerequis: string, private _titre: string,
    private _contenu: string, private _objectif: string, private _duree: number) {}

  public get prerequis(): string {
    return this._prerequis;
  }
  public set prerequis(_prerequis: string) {
    this._prerequis = _prerequis;
  }

  public get titre(): string {
    return this._titre;
  }
  public set titre(_titre: string) {
    this._titre = _titre;
  }
  public get contenu(): string {
    return this._contenu;
  }
  public set contenu(_contenu: string) {
    this._contenu = _contenu;
  }

  public get objectif(): string {
    return this._objectif;
  }
  public set objetif(_objectif: string) {
    this._objectif = _objectif;
  }

  public get id(): number {
    return this._id;
  }
  public set id(_id: number) {
    this._id = _id;
  }
  public get duree(): number {
    return this._duree;
  }
  public set duree(_duree: number) {
    this._duree = _duree;
  }
}
